pkg/controller/elasticsearch/certificates/transport/reconcile.go

```diff
@@ -39,8 +40,24 @@ func ReconcileTransportCertificatesSecrets(
 	rotationParams certificates.RotationParams,
 ) *reconciler.Results {
 	results := &reconciler.Results{}
+
+	// We must create transport certificates for the following StatefulSets:
+	// - the ones that still exist, even if they have been removed from the Spec
+	// - the ones that do not exist yet, but will be created in a later step of the reconciliation
+	actualStatefulSets, err := sset.RetrieveActualStatefulSets(c, k8s.ExtractNamespacedName(&es))
+	if err != nil {
+		return results.WithError(err)
+	}
+	ssets := make(map[string]struct{})
```

Comment:
We have a `StringSet` utility, which makes this a little bit better to read. I recently changed it to support an empty set to start with.